core/priority/prioritiser.go

```diff
@@ -392,28 +392,31 @@ func hashProto(msg proto.Message) ([32]byte, error) {
 }
 
 // sendResponse sends own response to any awaiting peers.
-func sendResponse(msgs map[string]received, recv received) {
-	if recv.Own { // Send our message to all waiting peers.
-		for _, other := range msgs {
-			if other.Response == nil {
-				continue
-			}
-			other.Response <- recv.Msg
+func sendResponse(msgs map[string]received) {
+	// Get own message
+	var own *pbv1.PriorityMsg
+	for _, recv := range msgs {
+		if !recv.Own {
+			continue
 		}
 
-		return
+		own = recv.Msg
+
+		break
 	}
 
-	if recv.Response == nil {
-		// This peer doesn't need a response
+	if own == nil {
+		// Own message not received yet
 		return
 	}
 
-	// Send own response to this peer.
-	for _, other := range msgs {
-		if !other.Own {
+	// Send own to any awaiting peers
+	for _, recv := range msgs {
+		if recv.Response == nil {
 			continue
 		}
-		recv.Response <- other.Msg
+
+		recv.Response <- own // Response channel has capacity of 1.
+		recv.Response = nil  // Mark channel as responded..
```

Comment:
why not `close(recv.Response)` to "Mark channel as responded"?

Reply:
I refactored the whole thing, should be more obvious now.